tasks/log_groups.yml

```diff
@@ -1,13 +1,12 @@
 ---
-
 - name: Ensure log groups {{ item.log_group_name }} exist 
   cloudwatchlogs_log_group:
     log_group_name: "{{ item.log_group_name }}"
-    retention: "{{item.log_retention|int}}"
-    tags: "{{item.log_group_tags}}"
-    aws_access_key: "{{ deploy_aws_access_key | default(aws_access_key | default(lookup('env','AWS_ACCESS_KEY_ID'))) }}"
-    aws_secret_key: "{{ deploy_aws_secret_key | default(aws_secret_key | default(lookup('env','AWS_SECRET_ACCESS_KEY'))) }}"
-    region: "{{ aws_region }}"
+    retention: "{{ item.log_retention|int }}"
+    tags: "{{ item.log_group_tags }}"
+    aws_access_key: "{{ aws_cwa_key_access | default(omit) }}"
```

Comment:
Not sure this is a good move, `aws_cwa_key_access`/`aws_cwa_key_secret`, other than being a terrible name, are the runtime credentials, this change forces them to have more permissions that are only required during the setup phase.

Reply:
I don't understand sorry. I've assumed these are only used in this role and not in `wg-ansible` at all. So, in this case, these variables become required by the role, so they can be set when the role is called or in wg-ansible inventories.

Reply:
See https://github.com/Webgains/ansible-role-amazon-cloudwatch-agent/blob/master/tasks/main.yml#L24:L37, that task sets up the credentials that the agent uses at run time, this task is run during instillation and need different permissions than the agent requires at run time. By linking the two together you are forcing the runtime credentials to have greater permissions than needed.

That being said this is a mess and needs to be removed, but probably better to do as a separate task as it will change a lot of places.

Reply:
So with the current use case what will happen is the credential file created on the remote will use the currently assumed aws_key and aws_secret instead of some shared key or a specific user's credentials.

If run via the common role it'll use `host_aws_access_key_id` and `host_aws_secret_access_key`